File: misc/clqz/1.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    count = next = 0
    visited = {}
    while(next<len(A) and next>=0):
        count += 1
        next += A[next]
        if next in visited:
            return -1
        visited[next] = True
    return count

File: misc/frht/1.go
package solution

// you can also use imports, for example:
// import "fmt"
// import "os"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

func Solution(A []int) int {
	count := 0
	next := 0
	visited := make(map[int]bool)
	for next < len(A) && next >= 0 {
		count++
		next += A[next]
		if _, ok := visited[next]; ok {
			return -1
		}
		visited[next] = true
	}
	return count
}
